track/__main__.py

```diff
@@ -48,31 +47,49 @@ def utc2localtime(utc):
+            user = contact.first_name+' '+contact.last_name
+        if printName == True:
+            print(user, file = dataFile)
+            print(user)
 
-    if isinstance(contact.status, UserStatusOffline):
-        if online != False:
-            online = False
-            print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline.')
-        elif last_offline != contact.status.was_online:
-            if last_offline is not None:
-                print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline after being online for short time.')
-            else:
+        if isinstance(contact.status, UserStatusOffline):
+            if online != False:
+                online = False
+                print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline.', file = dataFile)
                 print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline.')
-        last_offline = contact.status.was_online
-    elif isinstance(contact.status, UserStatusOnline):
-        if online != True:
-            online = True
-            print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went online.')
-    else:
-        if online != False:
-            online = False
-            print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went offline.')
-        last_offline = None
-    sleep(15)
+            elif last_offline != contact.status.was_online:
+                if last_offline is not None:
+                    print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline after being online for short time.', file = dataFile)
+                    print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline after being online for short time.')
+                else:
+                    print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline.', file = dataFile)
+                    print(f'={utc2localtime(contact.status.was_online).strftime(DATETIME_FORMAT)}: User went offline.')
+            last_offline = contact.status.was_online
+        elif isinstance(contact.status, UserStatusOnline):
+            if online != True:
+                online = True
+                print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went online.', file = dataFile)
+                print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went online.')
+        else:
+            if online != False:
+                online = False
+                print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went offline.', file = dataFile)
+                print(f'~{datetime.now().strftime(DATETIME_FORMAT)}: User went offline.')
+            last_offline = None
+        sleep(15)
+        printName = False
+        dataFile.flush()
```

Comment:
Doesn't seem useful to me. The file is closed one line below in the `finally` block.

Reply:
This is actually the only change I can really justify, without it there is no way to check the `.txt` for current changes without stopping the script, say for when you need to check from another PC, without using `tmux`